File: Kiip/Kiip-Android-SDK-v1.0.14-2/kiip-example/src/me/kiip/example/ExampleApplication.java
package me.kiip.example;

import me.kiip.api.Kiip;
import me.kiip.api.Kiip.RequestListener;
import me.kiip.api.KiipException;
import me.kiip.api.Resource;
import android.app.Application;
import android.util.Log;
import android.widget.Toast;

public class ExampleApplication extends Application {
    private static final String TAG = "example";

    private static final String KP_APP_KEY =    "0b56b49f621ad7f42fd85de7e461f9dd";
    private static final String KP_APP_SECRET = "ac3abfdf5cb86ce0febba0c8afd2744e";

    /** Called when the application is first created. */
    @Override
    public void onCreate() {
        super.onCreate();
        // The application has just launched, create Kiip instance.
        Kiip.init(this, KP_APP_KEY, KP_APP_SECRET);
    }

    public RequestListener<Resource> startSessionListener = new RequestListener<Resource>() {
        @Override
        public void onFinished(Kiip manager, Resource response) {
            if (response != null) {
                toast("Start Session Finished w/ Promo");
            } else {
                toast("Start Session Finished No Promo");
            }
            manager.showResource(response);
        }

        @Override
        public void onError(Kiip manager, KiipException error) {
            toast("Start Session Failed: " + "(" + error.getCode() + ") " + error.getMessage());
        }
    };

    public RequestListener<Resource> endSessionListener = new RequestListener<Resource>() {
        @Override
        public void onFinished(Kiip manager, Resource response) {
            toast("End Session Finished");
        }

        @Override
        public void onError(Kiip manager, KiipException error) {
            toast("End Session Failed: " + "(" + error.getCode() + ") " + error.getMessage());
        }
    };

    private void toast(String message) {
        Log.v(TAG, message);
        Toast.makeText(this, message, Toast.LENGTH_SHORT).show();
    }
}

File: Kiip/KiipSampleApplication/ExampleApplication.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Android.App;
using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Views;
using Android.Widget;
using ME.Kiip.Api;
using Android.Util;

namespace KiipSampleApplication
{
    [Application()]
    public class ExampleApplication : Application
    {
        private static readonly string TAG = "example";
        private static readonly string KP_APP_KEY = "xxx";
        private static readonly string KP_APP_SECRET = "yyy";

        public ME.Kiip.Api.Kiip.IRequestListener startSessionListener;
        public ME.Kiip.Api.Kiip.IRequestListener endSessionListener;

        public ExampleApplication(IntPtr handle, JniHandleOwnership transfer): base(handle, transfer)
        {

        }
    
        public override void OnCreate()
        {
            base.OnCreate();
            Kiip.Init(this, KP_APP_KEY, KP_APP_SECRET);
            startSessionListener = new StartSessionListener(this) as ME.Kiip.Api.Kiip.IRequestListener;
            endSessionListener = new EndSessionListener(this) as ME.Kiip.Api.Kiip.IRequestListener;

        }

        public void toast(string message)
        {
            Log.Verbose(TAG, message);
            Toast.MakeText(this, message, ToastLength.Short).Show();
        }

        public class StartSessionListener : Java.Lang.Object, ME.Kiip.Api.Kiip.IRequestListener
        {
            ExampleApplication example;

            public StartSessionListener(ExampleApplication example)
            {
                this.example = example;
            }

            public void OnError(Kiip p0, KiipException p1)
            {
                example.toast("End Session Failed: " + "(" + p1.Code + ") " + p1.Message);
            }

            public void OnFinished(Kiip p0, Java.Lang.Object p1)
            {
                //ME.Kiip.Api.Resource response = p1.JavaCast<ME.Kiip.Api.Resource>();
                //if (response != null)
                //{
                //    example.toast("Start Session Finished w/ Promo");
                //}
                //else
                //{
                //    example.toast("Start Session Finished No Promo");
                //}
                //p0.ShowResource(response);
            }
        }

        public class EndSessionListener : Java.Lang.Object, ME.Kiip.Api.Kiip.IRequestListener
        {
            ExampleApplication example;

            public EndSessionListener(ExampleApplication example)
            {
                this.example = example;
            }

            public void OnError(Kiip p0, KiipException p1)
            {
                example.toast("End Session Failed: " + "(" + p1.Code + ") " + p1.Message);
            }

            public void OnFinished(Kiip p0, Java.Lang.Object p1)
            {
                example.toast("End Session Finished");
            }
        }
    }
}